qa-tests-backend/src/test/groovy/ImageSignaturesTest.groovy

```diff
@@ -0,0 +1,270 @@
+        createdSignatureIntegrations.put(TEKTON, tektonSignatureIntegrationID)
+
+        // Signature integration "Unverifiable" which holds only the unverifiable cosign public key.
+        String unverifiableSignatureIntegrationID = createSignatureIntegration(
+                UNVERIFIABLE, UNVERIFIABLE_COSIGN_PUBLIC_KEY
+        )
+        assert unverifiableSignatureIntegrationID
+        createdSignatureIntegrations.put(UNVERIFIABLE, unverifiableSignatureIntegrationID)
+
+        // TODO(ROX-9996): Uncomment after we can handle multiple verification results.
+        /* Signature integration "Distroless+Tekton" which holds both distroless and tekton cosign public keys.
+        Map<String,String> mergedKeys = DISTROLESS_PUBLIC_KEY.clone() as Map<String, String>
+        mergedKeys.putAll(TEKTON_COSIGN_PUBLIC_KEY.entrySet())
+        String distrolessAndTektonSignatureIntegrationID = createSignatureIntegration(
+                DISTROLESS_AND_TEKTON, mergedKeys
+        )
+        assert distrolessAndTektonSignatureIntegrationID
+        createdSignatureIntegrations.put(DISTROLESS_AND_TEKTON, distrolessAndTektonSignatureIntegrationID)
+        */
+
+        // Create all required deployments.
+        orchestrator.batchCreateDeployments(DEPLOYMENTS)
+        for (Deployment deploymentID : DEPLOYMENTS) {
+            assert Services.waitForDeployment(deploymentID)
+        }
+
+        // Create the policy builders using the signature integration IDs.
+        List<Policy.Builder> policyBuilders = []
+        for (integrationName in INTEGRATION_NAMES) {
+            Policy.Builder builder = createPolicyBuilderWithSignatureCriteria(integrationName,
+                    [createdSignatureIntegrations.get(integrationName, "")])
+            assert builder
+            policyBuilders.add(builder)
+        }
+
+        // Create policies we use within tests.
+        for (policyBuilder in policyBuilders) {
+            Policy policy = policyBuilder.build()
+            String policyID = PolicyService.createNewPolicy(policy)
+            assert policyID
+            createdPolicyIds.add(policyID)
+        }
+    }
+
+    def cleanupSpec() {
+        // Delete all deployments.
+        for (Deployment deployment : DEPLOYMENTS) {
+            orchestrator.deleteDeployment(deployment)
+        }
+
+        // Delete all created policies.
+        for (policyID in createdPolicyIds) {
+            PolicyService.deletePolicy(policyID)
+        }
+
+        // Delete all created signature integrations.
+        for (signatureIntegration in createdSignatureIntegrations) {
+            SignatureIntegrationService.deleteSignatureIntegration(signatureIntegration.value)
+        }
+
+        orchestrator.deleteNamespace(SIGNATURE_TESTING_NAMESPACE)
+    }
+
+    @Unroll
+    @SuppressWarnings('LineLength')
+    @Category(Integration)
+    def "Check violations of policy '#policyName' for deployment '#deployment.name'"() {
```

Comment:
Would it make sense to also include the test case for several verifiers being enabled?

Reply:
What do you mean by `several verifiers being enabled`? An integration with multiple public keys? That's what currently is uncommented and will be added after [ROX-9996](https://issues.redhat.com/browse/ROX-9996) has landed.
Or is this something different you had in mind?

Reply:
I meant several separate integrations since they would produce different `verifierId`

Reply:
I am already creating three separate signature integrations, distroless tekton and unverifiable.